### PR TITLE
Template layouts

### PR DESCRIPTION
**Pull request by duncanmeech**
for /projects/combinatorial this implements list blocks rendering and the basics of editing constraints around frozen/fixed constructs

### CODE REVIEW

File: src/containers/graphics/views/layout.js [hunk 1]
```diff
@@ -12,6 +13,1962 @@ import invariant from 'invariant';
 const blockType = 'block';
 const roleType = 'role';
 
+const listStore = [
```
Comment (maxbates) [inline src/containers/graphics/views/layout.js]: oh god no. just import parts.js - the IDs will change soon anyway.

Comment (maxbates) [inline src/containers/graphics/views/layout.js]: but added a commit last week that puts them all in the store so you shouldnt need this anwyay

File: src/models/Instance.js [hunk 2]
```diff
@@ -24,7 +24,8 @@ export default class Instance {
     );
 
     if (process.env.NODE_ENV !== 'production') {
-      require('deep-freeze')(this);
+      console.warn();('Objects are not frozen for temporary testing.');
+      //require('deep-freeze')(this);
```
Comment (maxbates) [inline src/models/Instance.js]: Why do you need this?

File: src/containers/graphics/views/layout.js (hunk 1 above)
Comment (duncanmeech) [inline src/containers/graphics/views/layout.js]: Its just temporary...so you can ignore. I don't really want to keep PR's in a perpetual state of update while we finalize functionality. Why not merge this one and adjust in the inevitable next PR?

File: src/models/Instance.js (hunk 2 above)
Comment (duncanmeech) [inline src/models/Instance.js]: Was necessary when I was hacking in list blocks for testing. Needs to be removed, next commit.

File: src/containers/graphics/views/layout.js [hunk 3]
```diff
@@ -3,15 +3,23 @@ import Vector2D from '../geometry/vector2d';
...
 import kT from './layoutconstants';
 import objectValues from '../../../utils/object/values';
 import invariant from 'invariant';
+import partList from '../../../inventory/andrea/partList.json';
 
 // just for internal tracking of what type of block a node represents.
 const blockType = 'block';
 const roleType = 'role';
 
+// map the parts to a hash by part ID
```
Comment (maxbates) [inline src/containers/graphics/views/layout.js]: they are in the store now...

File: src/containers/graphics/views/layout.js [hunk 4]
```diff
@@ -514,12 +564,17 @@ export default class Layout {
...
     let nestedVertical = 0;
 
+    // additional height required by the tallest list on the row
+    let maxListHeight = 0;
+
     // width of first row is effected by parent block, so we have to track
     // which row we are on.
     let rowIndex = 0;
+    // display only non hidden blocks
+    const components = ct.components.filter(part => !this.partRule(part, 'hidden'));
```
Comment (maxbates) [inline src/containers/graphics/views/layout.js]: use block.isHidden() in case this changes

Comment (duncanmeech) [inline src/containers/graphics/views/layout.js]: fixed

File: src/containers/graphics/views/layout.js (hunk 3 above)
Comment (duncanmeech) [inline src/containers/graphics/views/layout.js]: fixed

**Pull request merged**